### ISSUE TITLE
Gradient visualization tool

### ISSUE THREAD

**Issue by old-school-kid** [keras-team/keras-cv#534]
**Short Description**
Incuding gradient based saliency map as a visualization tool. [This paper](https://arxiv.org/abs/1810.03292) (citations: 1015) goes through some of the visaulization tool. TL;DR : "Of the methods we tested, Gradients and GradCAM pass the sanity checks, while Guided BackProp and Guided GradCAM fail". 
 
**Papers**
https://arxiv.org/abs/1312.6034

**Implementation**
[tf-explain)(https://github.com/sicara/tf-explain/blob/master/tf_explain/core/vanilla_gradients.py) 

**Other Information**
Similar to th GradCAM tool (after it has been merged), I can work on this to create a tool for visualizing the gradients.

**Comment by LukeWood** [comment]
For now we've decided to keep the package scope more narrow.  Gradient visualization falls a bit outside our scope - so lets close this for now.
